collision.go

```diff
@@ -28,12 +29,46 @@ func (sc *SpaceComponent) Center(p Point) {
 }
 
 func (sc SpaceComponent) AABB() AABB {
```

Comment:
I skipped mentioning documentation on a couple of other exported functions to not be extremely repetitive but this function really deserves one. I can't tell what it does from its contents.